app/lib/features/pins/pages/pins_list_page.dart

```diff
@@ -0,0 +1,188 @@
+import 'dart:math';
+
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/toolkit/buttons/primary_action_button.dart';
+import 'package:acter/common/utils/routes.dart';
+import 'package:acter/common/widgets/add_button_with_can_permission.dart';
+import 'package:acter/common/widgets/empty_state_widget.dart';
+import 'package:acter/features/invite_members/providers/invite_providers.dart';
+import 'package:acter/features/pins/providers/pins_provider.dart';
+import 'package:acter/features/pins/widgets/pin_list_item.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:flutter_staggered_grid_view/flutter_staggered_grid_view.dart';
+import 'package:go_router/go_router.dart';
+
+class PinsListPage extends ConsumerStatefulWidget {
+  final String? spaceId;
+
+  const PinsListPage({super.key, this.spaceId});
+
+  @override
+  ConsumerState<PinsListPage> createState() => _AllPinsPageConsumerState();
+}
+
+class _AllPinsPageConsumerState extends ConsumerState<PinsListPage> {
+  final TextEditingController searchTextController = TextEditingController();
+  bool hasSearchTerm = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: _buildAppBar(),
+      body: _buildBody(),
+    );
+  }
+
+  AppBar _buildAppBar() {
+    return AppBar(
+      centerTitle: false,
+      title: Column(
+        mainAxisAlignment: MainAxisAlignment.start,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Text(L10n.of(context).pins),
+          _buildSpaceName(),
+        ],
+      ),
+      actions: [
+        AddButtonWithCanPermission(
+          canPermissionParam: widget.spaceId != null
+              ? (spaceId: widget.spaceId!, canString: 'CanPostPin')
+              : null,
+          onPressed: () => context.pushNamed(
+            Routes.actionAddPin.name,
+            queryParameters:
+                widget.spaceId != null ? {'spaceId': widget.spaceId} : {},
+          ),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildSpaceName() {
+    if (widget.spaceId != null) {
+      String spaceName =
+          ref.watch(roomDisplayNameProvider(widget.spaceId!)).valueOrNull ?? '';
+      return Text(
+        '($spaceName)',
+        overflow: TextOverflow.ellipsis,
+        style: Theme.of(context).textTheme.labelLarge,
+      );
+    } else {
+      return const SizedBox.shrink();
+    }
+  }
+
+  Widget _buildBody() {
+    AsyncValue<List<ActerPin>> pinList;
+
+    final searchValue = ref.watch(searchValueProvider) ?? '';
+    hasSearchTerm = searchValue.isNotEmpty;
```

Comment:
let's not have side-effects in the render-function please.

also ... why not just make the bool a getter function that just evaluates? like `get hasSearchTerm -> (ref.watch(searchValueProvider) ?? '').isNotEmpty`

Reply:
>  have side-effects in the render-function please

Sorry but getting your this point, can you explain it bit more?

> why not just make the bool a getter function that just evaluates

Yes, that's good idea.

Reply:
> Sorry but getting your this point, can you explain it bit more?

The render function is meant to be executed because of state changes. Thus, it should _not_ change state while executing. Setting variables that are then set the same way for longer than the render execution takes (e.g. in this case object variables) is such a state change that shouldn't be happening within the rendering function. 

Reasoning behind that is that you want to have a clean `event -> state change -> render` path but if the state can be, and is, modified by the `render` you make an unclean internal mismatch which makes it harder to reason about the code and when thing change or what state things are in at a certain point in time. To avoid such debugging difficulties, `render` functions are expected to be _side effect free_, as-in, they don't change the state, they just render it.

Reply:
(which is the paradigm that flutter generally puts forward and that riverpod made even more stringent. so if that happens _now_ it is most certainly just bad style ;) )

Reply:
Thanks @gnunicorn for bit of explain but still not sure I am fully understanding this thing.
However I did some changes in this commit. Can you please check and confirm if it is good or not.
https://github.com/acterglobal/a3/pull/1926/commits/8c8f42dde527740405e7053d6ad368cacdb11358

Reply:
yes, this is all good. in essence: do not change object variables while in `render`. your update does that now by removing that variable entirely.

Reply:
Ahh, now I got your point. Thanks.